client/allocrunner/alloc_runner_test.go

```diff
@@ -780,7 +780,7 @@ func TestAllocRunner_HandlesArtifactFailure(t *testing.T) {
 
 // Test that alloc runner kills tasks in task group when another task fails
 func TestAllocRunner_TaskFailed_KillTG(t *testing.T) {
-	alloc := mock.BatchAlloc()
+	alloc := mock.Alloc()
```

Comment:
I modified this existing test to change it to use a service job rather than batch so that it will get the  `DeploymentStatus` field. Could use feedback on whether this is the best way to test this.

Reply:
I think this is asserting reasonable things, but I think it would have passed before?

Restoring a Failed alloc with a Healthy DeploymentStatus and asserting `ar.AllocState().DeploymentStatus.IsHealthy()` would have failed before but worked now I think.

Reply:
@schmichael is there an existing test you could point me to as an example? I am not able to get this test's DeploymentStatus to become healthy through the mock driver.